DBM-Core/modules/TargetScanning.lua

```diff
@@ -140,7 +141,7 @@ do
 		--Perform normal scan criteria matching
 		elseif targetname and targetname ~= CL.UNKNOWN and (not targetFilter or (targetFilter and targetFilter ~= targetname)) then
 			if not IsInGroup() then scanTimes = 1 end--Solo, no reason to keep scanning, give faster warning. But only if first scan is actually a valid target, which is why i have this check HERE
-			if (isEnemyScan and UnitIsFriend("player", targetuid) or (onlyPlayers and not UnitIsPlayer("player", targetuid)) or mod:IsTanking(targetuid, bossuid)) and not isFinalScan then--On player scan, ignore tanks. On enemy scan, ignore friendly player. On Only player, ignore npcs and pets
+			if (isEnemyScan and UnitIsFriend("player", tostring(targetuid)) or (onlyPlayers and not UnitIsPlayer(targetuid)) or mod:IsTanking(targetuid, bossuid)) and not isFinalScan then--On player scan, ignore tanks. On enemy scan, ignore friendly player. On Only player, ignore npcs and pets
```

Comment:
What's the purpose of tostring here? The only occurances of targetuid is a string (check checking inside of the function returns)

Reply:
the relationship non-nil relationship between the return parameters of GetBossTarget above can't be encoded in the type system, so it doesn't know that targetname non nil implies that targetuid is also non-nil. i can change to check for targetuid in the if instead